### PR TITLE
Remove Eq from `universeNotEmpty`

### PR DESCRIPTION
**Pull request by mjgpy3**
I'm not sure if this is right or even desirable. I figured the fewer constraints the more potential mileage.

### HLint

- [X] I've changed the exposed interface (add new reexports, remove reexports, rename reexported things, etc.).
  - [ ] I've updated [`hlint.dhall`](https://github.com/kowainik/relude/blob/master/hlint/hlint.dhall) accordingly to my changes (add new rules for the new imports, remove old ones, when they are outdated, etc.).
Is this necessary for this change?
  - [ ] I've generated the new `.hlint.yaml` file (see [this instructions](https://github.com/kowainik/relude#generating-hlintyaml)).
Is this necessary for this change?
### General

- [X] I've updated the [CHANGELOG](https://github.com/kowainik/relude/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [X] All new and existing tests pass.

I've only run `stack test`. Is there a way to execute doc examples?

- [X] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [X] I've used the [`stylish-haskell` file](https://github.com/kowainik/relude/blob/master/.stylish-haskell.yaml).
- [X] My change requires the documentation updates.
  - [X] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.

### CODE REVIEW

File: src/Relude/Extra/Enum.hs [hunk 1]
```diff
@@ -67,10 +67,8 @@ Singleton :| []
 
 @since 0.7.0.0
 -}
-universeNonEmpty :: forall a . (Bounded a, Enum a, Eq a) => NonEmpty a
-universeNonEmpty
-    | minBound @a == maxBound = minBound :| []
-    | otherwise = minBound :| [succ minBound .. maxBound]
+universeNonEmpty :: forall a . (Bounded a, Enum a) => NonEmpty a
+universeNonEmpty = minBound :| drop 1 universe
```
Comment (mjgpy3) [inline src/Relude/Extra/Enum.hs:71]: I'd be happy to rewrite in a self-contained way (e.g. explicit inline of `universe`) if that's preferred.

Comment (vrom911) [reply src/Relude/Extra/Enum.hs:71]: I think that this is pretty neat! Thanks 👍🏼

**Review by vrom911 — APPROVED** [review]
This is a nice idea! I fully support this, thank you 💯

**Review by chshersh — APPROVED** [review]
Thanks, looks neat! 👍🏻

**Pull request merged**